akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -87,6 +87,7 @@ abstract class ModeledCustomHeaderCompanion[H <: ModeledCustomHeader[H]] {
     case _               ⇒ None
   }
 
+  final implicit val implicitlyLocatableCompanion: ModeledCustomHeaderCompanion[H] = this
```

Comment:
This allows us to implicitly find the companion, if we know the type of a header (as given `T`, implicit search scope includes the `T` object, in which we find this)